hnn_core/network.py

```diff
@@ -35,27 +37,54 @@ def read_spikes(fname, gid_dict=None):
     spike_times = []
     spike_gids = []
     spike_types = []
+    spike_tstart = []
+    spike_tstop = []
     for file in sorted(glob(fname)):
         spike_trial = np.loadtxt(file, dtype=str)
         spike_times += [list(spike_trial[:, 0].astype(float))]
         spike_gids += [list(spike_trial[:, 1].astype(int))]
 
         # Note that legacy HNN 'spk.txt' files don't contain a 3rd column for
-        # spike type. If reading a legacy version, validate that a gid_dict is
-        # provided.
-        if spike_trial.shape[1] == 3:
+        # spike type, or a 4th and 5th column for tstart and tstop. If reading
+        # a legacy version, validate that gid_dict, tstart, and tstop provided.
```

Comment:
why another column? I would have added it in the beginning before starting the rows since it's going to be the same for all spikes. See also related discussion [here](https://github.com/jonescompneurolab/hnn-core/pull/83#discussion_r478705027) and in #159

Reply:
The main motivation was to make it easy to check the version of the `spk.txt` file (since there are already a variable number of columns between the versions). 

What do you think of adding a string to the top left entry that will indicate fixed parameters. Something like `f.write('{}\t{:.3f}\t{:.3f}\n'.format('Fixed Values', float(self._tstart), 'float(self._tstop)')`?

That way the logic just involved checking the string and skipping rows if it is present.

Reply:
I was thinking of using [`np.loadtxt`](https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html) with comments="#". Then you can have two lines in the beginning which specify tstart and tstop. But how will this work for multiple trials? Are we going to check that tstart and tstop is same across trials? Will each trial be stored in different files or the same?

I feel we shouldn't develop a new format ad hoc and spend some time to think what should be the ideal format and what are the pros and cons of different formats. Probably a good discussion topic for an HNN meeting or even during the hackathon.

Reply:
Tagging @blakecaldwell and @rythorpe for suggestions

Reply:
We have needed to decide on a way to provide annotations in hnn-core data files for awhile, so I agree with @jasmainak that this might be a good time to start thinking about how to do this systematically. I'm unfamiliar with best-practices in this area, but I like the idea of adding annotations as lines to the top of each file. We could even supply a `legacy` boolean argument to the `read_spikes()` function in order to simplify the logic that handles identification of current vs. legacy versions.